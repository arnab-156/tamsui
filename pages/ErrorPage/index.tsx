import React from 'react';
import { Link } from 'react-router-dom';

import Layout from 'pages/Layout';

export default function ErrorPage() {
  return (
    <Layout>
      <h1>An Error Occurred!</h1>
      <Link to="/">Home</Link>
    </Layout>
  );
}
