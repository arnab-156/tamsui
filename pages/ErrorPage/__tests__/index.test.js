import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ErrorPage from '../index';

function RouterWrappedErrorPage() {
  return (
    <MemoryRouter>
      <ErrorPage />
    </MemoryRouter>
  );
}

describe('ErrorPage Component', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<RouterWrappedErrorPage />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('contains the error text', () => {
    render(<RouterWrappedErrorPage />);

    expect(screen.getByRole('heading')).toHaveTextContent('An Error Occurred!');
  });
});
