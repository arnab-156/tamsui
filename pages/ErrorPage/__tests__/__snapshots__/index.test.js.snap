// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component matches snapshot 1`] = `
<section
  className="page"
>
  <nav
    className="navbar"
  >
    <div
      className="contents"
    >
      <div
        className="segment home"
      >
        <a
          className="menuLink"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Tamsui logo"
            className="logo"
            src="/static/images/logo.webp"
          />
          <span>
            Tamsui
          </span>
        </a>
      </div>
      <div
        className="segment pages"
      >
        <a
          className="menuLink"
          href="/counter"
          onClick={[Function]}
        >
          Counter
        </a>
      </div>
      <div
        className="segment social"
      >
        <a
          href="https://github.com/chichiwang/tamsui"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Github logo"
            className="logo"
            src="/static/images/github-mark.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="content"
  >
    <h1>
      An Error Occurred!
    </h1>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </main>
</section>
`;
